query-engine/connector-test-kit-rs/qe-setup/src/lib.rs

```diff
@@ -35,30 +54,65 @@ fn parse_configuration(datamodel: &str) -> ConnectorResult<(Datasource, String,
     Ok((source, url, preview_features))
 }
 
+/// Database setup for connector-test-kit-rs with Driver Adapters.
+/// If the external driver adapter requires a migration by means of the JavaScript runtime
+/// (rather than just the Schema Engine), this function will call [`ExternalInitializer::init_with_migration`].
+/// Otherwise, it will call [`ExternalInitializer::init`], and then proceed with the standard
+/// setup based on the Schema Engine.
+pub async fn setup_external<'a, EI>(
+    driver_adapter: DriverAdapter,
+    initializer: EI,
+    db_schemas: &[&str],
+) -> ConnectorResult<()>
+where
+    EI: ExternalInitializer<'a> + ?Sized,
+{
+    let prisma_schema = initializer.datamodel();
+    let (source, url, _preview_features) = parse_configuration(prisma_schema)?;
+
+    if driver_adapter == DriverAdapter::D1 {
+        // 1. Compute the diff migration script.
+        std::fs::remove_file(source.url.as_literal().unwrap().trim_start_matches("file:")).ok();
+        let mut connector = sql_schema_connector::SqlSchemaConnector::new_sqlite();
+        let migration_script = crate::diff(prisma_schema, url, &mut connector).await?;
+
+        // 2. Tell JavaScript to take care of the schema migration.
+        //    This results in a JSON-RPC call to the JS runtime.
+        //    The JSON-RPC machinery is defined in the `[query-tests-setup]` crate, and it
+        //    implements the `ExternalInitializer<'a>` trait.
+        initializer
+            .init_with_migration(migration_script)
+            .await
+            .map_err(|err| ConnectorError::from_msg(format!("Error migrating with D1 adapter: {}", err)))?;
+
+        // TODO: consider adding a busy loop until JavaScript acknowledges the migration is complete.
```

Comment:
Is it still needed?

Reply:
Not anymore! I've fixed it by merging initialisation + migration in the JSON-RPC `initializeSchema` method.